### PR TITLE
Update PHPStan level to 5

### PR DESCRIPTION
**Pull request by westonruter**
See #775.

This fixes the following PHPStan issues:

<details><summary>[ERROR] Found 11 errors</summary>

```
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/includes/server-timing/perflab-server-timing-tests.php                                                                                                                                   
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  111    Parameter #2 $args of method Perflab_Server_Timing::register_metric() expects array{measure_callback: callable(): mixed, access_cap: string}, array{access_cap: 'exist'} given.                
         💡 Array does not have offset 'measure_callback'.                                                                                                                                              
  122    Parameter #2 $args of method Perflab_Server_Timing::register_metric() expects array{measure_callback: callable(): mixed, access_cap: string}, array{measure_callback: '__return_null'} given.  
         💡 Array does not have offset 'access_cap'.                                                                                                                                                    
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   tests/includes/site-health/audit-autoloaded-options/audit-autoloaded-options-test.php                                                                             
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  27     Parameter #1 $tests of function perflab_aao_add_autoloaded_options_test expects array{direct: array<string, array{label: string, test: string}>}, array{} given.  
         💡 Array does not have offset 'direct'.                                                                                                                           
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------------------------------------------ 
  Line   tests/includes/site-health/audit-enqueued-assets/audit-enqueued-assets-helper-test.php                      
 ------ ------------------------------------------------------------------------------------------------------------ 
  93     Parameter #1 $resource_url of function perflab_aea_get_path_from_resource_url expects string, false given.  
 ------ ------------------------------------------------------------------------------------------------------------ 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/includes/site-health/audit-enqueued-assets/audit-enqueued-assets-test.php                                                                                
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  168    Parameter #1 $tests of function perflab_aea_add_enqueued_assets_test expects array{direct: array<string, array{label: string, test: string}>}, array{} given.  
         💡 Array does not have offset 'direct'.                                                                                                                        
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------- 
  Line   tests/plugins/auto-sizes/auto-sizes-test.php                                      
 ------ ---------------------------------------------------------------------------------- 
  49     Parameter #3 $icon of function wp_get_attachment_image expects bool, null given.  
  61     Parameter #3 $icon of function wp_get_attachment_image expects bool, null given.  
 ------ ---------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------ 
  Line   tests/plugins/dominant-color-images/dominant-color-test.php                         
 ------ ------------------------------------------------------------------------------------ 
  237    Parameter #3 $icon of function wp_get_attachment_image expects bool, string given.  
 ------ ------------------------------------------------------------------------------------ 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/plugins/webp-uploads/image-edit-tests.php                                                                                              
 ------ --------------------------------------------------------------------------------------------------------------------------------------------- 
  400    Parameter #2 $hashed_size_name of method PerformanceLab\Tests\TestCase\ImagesTestCase::assertSizeNameIsHashed() expects string, null given.  
 ------ --------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------ 
  Line   tests/plugins/webp-uploads/load-tests.php                                                                         
 ------ ------------------------------------------------------------------------------------------------------------------ 
  653    Parameter #1 $needle of method PHPUnit\Framework\Assert::assertStringContainsString() expects string, int given.  
  654    Parameter #1 $needle of method PHPUnit\Framework\Assert::assertStringContainsString() expects string, int given.  
 ------ ------------------------------------------------------------------------------------------------------------------ 
```

</details> 

Please review individual commits for changes.

### CODE REVIEW

File: tests/includes/server-timing/perflab-server-timing-tests.php [hunk 1]
```diff
@@ -108,6 +108,7 @@ public function test_register_metric_requires_measure_callback() {
 
 		$this->server_timing->register_metric(
 			'metric-without-measure-callback',
+			// @phpstan-ignore-next-line
```
Comment (mukeshpanchal27) [inline tests/includes/server-timing/perflab-server-timing-tests.php]: Could you add note here why we ignore it?

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks for the PR. LGTM.

File: tests/includes/server-timing/perflab-server-timing-tests.php (hunk 1 above)
Comment (westonruter) [reply tests/includes/server-timing/perflab-server-timing-tests.php]: Good idea. Added in c588e588

File: phpstan.neon.dist [hunk 2]
```diff
@@ -1,8 +1,9 @@
 includes:
 	- phar://phpstan.phar/conf/bleedingEdge.neon
 parameters:
-	level: 4
+	level: 5
 	treatPhpDocTypesAsCertain: false
+	rememberPossiblyImpureFunctionValues: false
```
Comment (thelovekesh) [inline phpstan.neon.dist:6]: I think it would be better if it had some doc reference on why this option was added?

**Review by thelovekesh — APPROVED** [review]
LGTM!

File: phpstan.neon.dist (hunk 2 above)
Comment (westonruter) [reply phpstan.neon.dist:6]: Oops. Merged before I saw this. Yes, good idea. Added in 0cd8565b7338cfa12753e70dc894902419db3805. I'll add a commit to https://github.com/WordPress/performance/pull/1203 to explain.

Comment (westonruter) [reply phpstan.neon.dist:6]: Fixed in b86637e72644ff5e09d3932a87c3c2de8ae9baa5

**Pull request merged**